canonicalwebteam/blog/django/views.py

```diff
@@ -86,6 +90,68 @@ def group(request, slug, template_path):
     return render(request, template_path, context)
 
 
+def archives(request, template_path="blog/archives.html"):
+    try:
+        page_param = request.GET.get("page", default="1")
+        category_param = request.GET.get("category", default="")
+        group_param = request.GET.get("group", default="")
+        month_param = request.GET.get("month", default="")
+        year_param = request.GET.get("year", default="")
+
+        group = None
+        group_id = ""
+        if group_param != "":
+            group = api.get_group_by_slug(group_param)
+            group_id = group["id"]
+
+        category_id = ""
+        if category_param != "":
+            category = api.get_category_by_slug(category_param)
+            category_id = category["id"]
+
+        month = ""
+        if month_param != "":
+            month = int(month_param)
+
+        year = ""
+        if year_param != "":
+            year = int(year_param)
+
+        if year != "":
+            after = datetime(year=year, month=1, day=1)
+            before = datetime(year=year, month=12, day=31)
+            if month != "":
+                after = datetime(year=year, month=month, day=1)
+                before = after + relativedelta(months=1)
+            else:
+                after = datetime(year=year, month=1, day=1)
+                before = after + relativedelta(years=1)
+
+        articles, total_pages = api.get_articles(
+            tags=tag_ids,
+            tags_exclude=excluded_tags,
+            page=page_param,
+            groups=[group_id],
+            categories=[category_id],
+            after=after,
+            before=before,
+        )
+
+    except Exception as e:
+        return HttpResponse("Error: " + e, status=502)
+
+    if group:
```

Comment:
Can we move 143-152 inside the try block?

Reply:
Works :)